LearningMissionSimulation/PlaygroundGarush.cs

```diff
@@ -93,28 +116,42 @@ List<Module> GetModuleList()
 
         public void AssignModulesToStudents()
         {
+            string action = "Assing Modules to students";
+            int count = 0;
+            ReportHeader(actionName: action);
             foreach (var student in studentList)
             {
                 student.CompletedModuleList = GetModuleList();
                 Console.WriteLine(student);
             }
+
+            ReportSummary(actionName: action, count: count);
+            ReportFooter(actionName: action);
         }
 
         public void CreateSubjects(int subjectCount)
         {
+            string action = "Subject generation";
+            ReportHeader(actionName: action);
             int i = 0;
             while (i < subjectCount)
-            {
+            {               
                 Subject subject = ObjectGenerator.GenerateSubject();
                 subjectList.Add(subject);
                 subjectIdList.Add(subject.Id);
                 Console.WriteLine("\ncreate subject {0}\n", subject);
                 i++;
+                ReportItem(itemName: subject.ToString(), actionName: action, itemIndex: i);
             }
+
+            ReportSummary(actionName: action, count:subjectCount);
+            ReportFooter(actionName: action);
         }
 
         public void CreateModules(int moduleCount)
         {
+            string action = "Subject generation";
+            ReportHeader(actionName: action);
             int i = 0;
             while (i < moduleCount)
```

Comment:
You have to check also 
subjectIdList.Count > 0, otherwise you well run into an exception line 158. If subjectList.Count==0 you have to report an error.

Reply:
It is not handled, please fix.